Recompile error for python

**Describe the Bug**
I followed the steps described [here](https://cyberbotics.com/doc/guide/using-python#windows-installation). MSYS2 has all requirements. However I get the following error after issuing `make` command
```
swig -c++ -python -outdir "C:/Program\ Files/Webots/lib/python36/" -I"C:/Program\ Files/Webots/include/controller/cpp" -I"C:/Program\ Files/Webots/include/controller/c" -o controller36.cpp controller.i
controller.i:115: Error: Unable to find 'webots/Device.hpp'
controller.i:121: Error: Unable to find 'webots/Accelerometer.hpp'
controller.i:127: Error: Unable to find 'webots/Brake.hpp'
controller.i:154: Error: Unable to find 'webots/camera_recognition_object.h'
controller.i:299: Error: Unable to find 'webots/Camera.hpp'
controller.i:307: Error: Unable to find 'webots/Compass.hpp'
controller.i:313: Error: Unable to find 'webots/Connector.hpp'
controller.i:367: Error: Unable to find 'webots/ImageRef.hpp'
controller.i:368: Error: Unable to find 'webots/Display.hpp'
controller.i:374: Error: Unable to find 'webots/DistanceSensor.hpp'
controller.i:385: Error: Unable to find 'webots/Emitter.hpp'
controller.i:396: Error: Unable to find 'webots/Field.hpp'
controller.i:402: Error: Unable to find 'webots/GPS.hpp'
controller.i:408: Error: Unable to find 'webots/Gyro.hpp'
controller.i:414: Error: Unable to find 'webots/InertialUnit.hpp'
controller.i:420: Error: Unable to find 'webots/Joystick.hpp'
controller.i:426: Error: Unable to find 'webots/Keyboard.hpp'
controller.i:432: Error: Unable to find 'webots/LED.hpp'
controller.i:440: Error: Unable to find 'webots/lidar_point.h'
controller.i:497: Error: Unable to find 'webots/Lidar.hpp'
controller.i:505: Error: Unable to find 'webots/LightSensor.hpp'
controller.i:511: Error: Unable to find 'webots/utils/Motion.hpp'
controller.i:517: Error: Unable to find 'webots/Motor.hpp'
controller.i:542: Error: Unable to find 'webots/mouse_state.h'
controller.i:544: Error: Unable to find 'webots/Mouse.hpp'
controller.i:569: Error: Unable to find 'webots/Node.hpp'
controller.i:575: Error: Unable to find 'webots/Pen.hpp'
controller.i:581: Error: Unable to find 'webots/PositionSensor.hpp'
controller.i:606: Error: Unable to find 'webots/radar_target.h'
controller.i:625: Error: Unable to find 'webots/Radar.hpp'
controller.i:686: Error: Unable to find 'webots/RangeFinder.hpp'
controller.i:698: Error: Unable to find 'webots/Receiver.hpp'
controller.i:706: Error: Unable to find 'webots/Skin.hpp'
controller.i:712: Error: Unable to find 'webots/Speaker.hpp'
controller.i:718: Error: Unable to find 'webots/TouchSensor.hpp'
controller.i:1009: Error: Unable to find 'webots/Robot.hpp'
controller.i:1015: Error: Unable to find 'webots/DifferentialWheels.hpp'
controller.i:1021: Error: Unable to find 'webots/Supervisor.hpp'
make: *** [Makefile:99: controller36.cpp] Error 1
```
All help is appreciated. Also do I have to recompile to use 3.6.5 instead of 3.7 version of python?

Thanks @omichel for doing that fix,
Apparently, there several bugs from doing the compilation on Windows.

```powershell
swig -c++ -python -outdir C:/Program\ Files/Webots/lib/python36/ -IC:/Program\ Files/Webots/include/controller/cpp -IC:/Program\ Files/Webots/include/controller/c -o controller36.cpp controller.i
g++ -c -O -Wall -DMS_WIN64 -D_hypot=hypot -Wno-stringop-truncation -IC:/Program\ Files/Webots/include/controller/cpp -IC:/Program\ Files/Webots/include/controller/c -I"/c/Python36/include" controller36.cpp -o controller36.o
In file included from /c/Python36/include/Python.h:53:0,
                 from controller36.cpp:176:
/c/Python36/include/pyport.h:686:2: error: #error "LONG_BIT definition appears wrong for platform (bad gcc/glibc config?)."
 #error "LONG_BIT definition appears wrong for platform (bad gcc/glibc config?)."
  ^~~~~
In file included from /c/Python36/include/Python.h:136:0,
                 from controller36.cpp:176:
/c/Python36/include/fileutils.h:41:5: error: '__int64' does not name a type; did you mean '__int64_t'?
     __int64 st_size;
     ^~~~~~~
     __int64_t
In file included from /usr/include/sys/stat.h:22:0,
                 from /c/Python36/include/pyport.h:227,
                 from /c/Python36/include/Python.h:53,
                 from controller36.cpp:176:
/c/Python36/include/fileutils.h:42:12: error: expected ';' at end of member declaration
     time_t st_atime;
            ^
/c/Python36/include/fileutils.h:42:12: error: expected unqualified-id before '.' token
     time_t st_atime;
            ^
/c/Python36/include/fileutils.h:44:12: error: expected ';' at end of member declaration
     time_t st_mtime;
            ^
/c/Python36/include/fileutils.h:44:12: error: expected unqualified-id before '.' token
     time_t st_mtime;
            ^
/c/Python36/include/fileutils.h:46:12: error: expected ';' at end of member declaration
     time_t st_ctime;
            ^
/c/Python36/include/fileutils.h:46:12: error: expected unqualified-id before '.' token
     time_t st_ctime;
            ^
cc1plus: warning: unrecognized command line option '-Wno-stringop-truncation'
make: *** [Makefile:111: controller36.o] Error 1
```

![2019-01-07_20-46-35](https://user-images.githubusercontent.com/5664750/50787161-4dcede80-12be-11e9-819d-1f9ef9a12a4b.png)

Moreover, MSYS2 doesn't have any repository for 3.6 and I am totally ignorant of how to make it aware of the version I have on Windows. PowerShell seems solving this but apparently compilation parameters still hold values that are inadequate to windows.

--- a/Makefile
+++ b/Makefile
@@ -105,11 +105,11 @@ webots_target: webots_dependencies
 	@+echo "#"; echo "# * webots (core) *"; echo "#"
 	@+make --silent -C src/webots $(TARGET)
 	@+echo "#"; echo "# * libController *"; echo "#"
-	@+make --silent -C src/lib/Controller $(TARGET) WEBOTS_HOME=$(WEBOTS_HOME)
+	@+make --silent -C src/lib/Controller $(TARGET) WEBOTS_HOME="$(WEBOTS_HOME)"
 	@+echo "#"; echo "# * resources *";
-	@+make --silent -C resources $(TARGET) WEBOTS_HOME=$(WEBOTS_HOME)
+	@+make --silent -C resources $(TARGET) WEBOTS_HOME="$(WEBOTS_HOME)"
 	@+echo "#"; echo "# * projects *";
-	@+make --silent -C projects $(TARGET) WEBOTS_HOME=$(WEBOTS_HOME)
+	@+make --silent -C projects $(TARGET) WEBOTS_HOME="$(WEBOTS_HOME)"
 
 webots_dependencies:
 	@+echo "#"; echo "# * dependencies *"; echo "#"

--- a/projects/default/controllers/ros/Makefile
+++ b/projects/default/controllers/ros/Makefile
@@ -50,23 +50,19 @@ endif
 
 $(CXX_SOURCES:.cxx=.d): $(GENERATED_HEADERS) ROS_HEADERS include/templateHeader.h include/templateRequest.h include/templateResponse.h
 
-ifndef CURDIR
- CURDIR = $(shell pwd)
-endif
-
 $(GENERATED_HEADERS): $(SERVICE_HEADERS) $(MESSAGE_HEADERS)
 
-include/messages/%.h: $(CURDIR)/include/msg/%.msg include/templateHeader.h include/templateRequest.h include/templateResponse.h headersFromMSG.py headersGenerator.py
+include/messages/%.h: include/msg/%.msg include/templateHeader.h include/templateRequest.h include/templateResponse.h headersFromMSG.py headersGenerator.py
 	@echo "# generating message header" $(notdir $<)
 	$(SILENT)python headersFromMSG.py $<
-	$(SILENT)mkdir -p $(CURDIR)/include/messages
-	$(SILENT)cp $(CURDIR)/include/webots_ros/$(basename $(notdir $<)).h $(CURDIR)/include/messages/$(basename $(notdir $<)).h
+	$(SILENT)mkdir -p include/messages
+	$(SILENT)cp include/webots_ros/$(basename $(notdir $<)).h include/messages/$(basename $(notdir $<)).h
 
-include/services/%.h: $(CURDIR)/include/srv/%.srv include/templateHeader.h include/templateRequest.h include/templateResponse.h headersFromSRV.py headersGenerator.py
+include/services/%.h: include/srv/%.srv include/templateHeader.h include/templateRequest.h include/templateResponse.h headersFromSRV.py headersGenerator.py
 	@echo "# generating service header" $(notdir $<)
 	$(SILENT)python headersFromSRV.py $<
-	$(SILENT)mkdir -p $(CURDIR)/include/services
-	$(SILENT)cp $(CURDIR)/include/webots_ros/$(basename $(notdir $<)).h $(CURDIR)/include/services/$(basename $(notdir $<)).h
+	$(SILENT)mkdir -p include/services
+	$(SILENT)cp include/webots_ros/$(basename $(notdir $<)).h include/services/$(basename $(notdir $<)).h
 
 ROS_HEADERS:
 	+@make -s -C include $(MAKECMDGOALS)

--- a/projects/default/controllers/ros/Makefile
+++ b/projects/default/controllers/ros/Makefile
@@ -71,11 +71,11 @@ include/services/%.h: $(CURDIR)/include/srv/%.srv include/templateHeader.h inclu
 ROS_HEADERS:
 	+@make -s -C include $(MAKECMDGOALS)
 
-INCLUDE = -isystem "$(WEBOTS_HOME_PATH)/projects/default/controllers/ros/include"
+INCLUDE = -isystem $(WEBOTS_HOME_PATH)/projects/default/controllers/ros/include
 
 # include ros libraries
 
-LIBRARIES += -L"$(WEBOTS_HOME_PATH)/projects/default/libraries/ros" -lroscpp -lrosconsole -lroscpp_serialization -lrostime
+LIBRARIES += -L$(WEBOTS_HOME_PATH)/projects/default/libraries/ros -lroscpp -lrosconsole -lroscpp_serialization -lrostime
 ifeq ($(OSTYPE),windows)
  LIBRARIES += -lws2_32
  ifeq ($(MAKECMDGOALS),debug)
@@ -88,7 +88,7 @@ ifeq ($(OSTYPE),windows)
 endif
 
 ifeq ($(OSTYPE),linux)
- LIBRARIES += -Wl,-rpath,"$(WEBOTS_HOME_PATH)/projects/default/libraries/ros"
+ LIBRARIES += -Wl,-rpath,$(WEBOTS_HOME_PATH)/projects/default/libraries/ros
 endif
 
 ifeq ($(OSTYPE),darwin)

--- a/projects/default/libraries/vehicle/java/Makefile
+++ b/projects/default/libraries/vehicle/java/Makefile
@@ -15,13 +15,14 @@
 space :=
 space +=
 WEBOTS_HOME_PATH=$(subst $(space),\ ,$(strip $(subst \,/,$(WEBOTS_HOME))))
+WEBOTS_HOME_PATH_NO_ESCAPE=$(subst \,/,$(WEBOTS_HOME))
 include $(WEBOTS_HOME_PATH)/resources/Makefile.os.include
 
 WEBOTS_LIBRARY = 1
 MODULE_NAME = vehicle
 
 INTERFACE = $(MODULE_NAME).i
-SOURCE_HEADERS = $(wildcard $(WEBOTS_HOME_PATH)/include/controller/cpp/webots/*.hpp)
+SOURCE_HEADERS = $(substr $(WEBOTS_HOME_PATH_NO_ESCAPE),$(WEBOTS_HOME_PATH),$(wildcard $(WEBOTS_HOME_PATH)/include/controller/cpp/webots/*.hpp))
 JAVA_FILES = Driver.java Car.java vehicle.java vehicleJNI.java
 
 WRAPPER = $(MODULE_NAME).cpp
@@ -73,7 +74,7 @@ all release debug profile: $(JAR) $(LIBRARY)
 
 $(JAR): JAVA_CLASSES
 	@mkdir -p $(WEBOTS_HOME_PATH)/lib/java
-	@jar cfv $@ $(MODULE_PATH)/*.class > /dev/null
+	@jar cfv "$@" $(MODULE_PATH)/*.class > /dev/null
 
 JAVA_CLASSES: $(JAVA_SOURCES)
 	@javac -target 1.8 -classpath $(WEBOTS_HOME_PATH)/lib/java/Controller.jar $(JAVA_SOURCES) -d .
@@ -85,19 +86,19 @@ $(WRAPPER): $(INTERFACE) $(SOURCE_HEADERS)
 	@swig -c++ -java -package $(subst /,.,$(MODULE_PATH)) -outdir $(SWIG_OUTPUT) -I$(WEBOTS_HOME_PATH)/resources/languages/java/ $(INCLUDES) -o $@ $<
 
 $(WRAPPER_OBJECT): $(WRAPPER)
-	@$(CXX) $(CFLAGS2) $(INCLUDES) $(JAVA_INCLUDES) $< -o $@
+	@$(CXX) $(CFLAGS2) $(INCLUDES) $(JAVA_INCLUDES) $< -o "$@"
 
 %.so: $(WRAPPER_OBJECT)
 	@mkdir -p $(WEBOTS_HOME_PATH)/lib/java
-	@$(CXX) -shared $< $(LIB) -o $@
+	@$(CXX) -shared $< $(LIB) -o "$@"
 
 %.dll: $(WRAPPER_OBJECT)
 	@mkdir -p $(WEBOTS_HOME_PATH)/lib/java
-	@$(CXX)  -shared -mwindows -Wl,--add-stdcall-alias -Wl,--enable-auto-import -O -lm $< $(LIB) -o $@
+	@$(CXX) -shared -mwindows -Wl,--add-stdcall-alias -Wl,--enable-auto-import -O -lm $< $(LIB) -o "$@"
 
 %.jnilib: $(WRAPPER_OBJECT)
 	@mkdir -p $(WEBOTS_HOME_PATH)/lib/java
-	@$(CXX) $(CFLAGS1) -bundle $< $(LIB) -o $@
+	@$(CXX) $(CFLAGS1) -bundle $< $(LIB) -o "$@"
 endif
 
 clean:

--- a/projects/default/libraries/vehicle/java/Makefile
+++ b/projects/default/libraries/vehicle/java/Makefile
@@ -27,7 +27,7 @@ JAVA_FILES = Driver.java Car.java vehicle.java vehicleJNI.java
 WRAPPER = $(MODULE_NAME).cpp
 WRAPPER_OBJECT = $(WRAPPER:.cpp=.o)
 JAR = $(WEBOTS_HOME_PATH)/lib/java/$(MODULE_NAME).jar
-INCLUDES = -I"$(WEBOTS_HOME_PATH)/include/controller/c/" -I"$(WEBOTS_HOME_PATH)/include/controller/cpp/"
+INCLUDES = -I$(WEBOTS_HOME_PATH)/include/controller/c/ -I$(WEBOTS_HOME_PATH)/include/controller/cpp/
 
 SWIG_OUTPUT  = output
 MODULE_PATH  = com/cyberbotics/webots/controller/vehicle
@@ -46,7 +46,7 @@ ifeq ($(OSTYPE),windows)
 JAVA_INCLUDES = -I"$(JAVA_HOME)/include" -I"$(JAVA_HOME)/include/win32"
 CFLAGS2       = -c -Wall -Wno-unused-function -O
 LIBRARY       = $(WEBOTS_HOME_PATH)/lib/java/vehicle.dll
-LIB           = -L"$(WEBOTS_HOME_PATH)/msys64/mingw64/bin"
+LIB           = -L$(WEBOTS_HOME_PATH)/msys64/mingw64/bin
 endif
 
 ifeq ($(OSTYPE),darwin)
@@ -76,7 +76,7 @@ $(JAR): JAVA_CLASSES
 	@jar cfv $@ $(MODULE_PATH)/*.class > /dev/null
 
 JAVA_CLASSES: $(JAVA_SOURCES)
-	@javac -target 1.8 -classpath "$(WEBOTS_HOME_PATH)/lib/java/Controller.jar" $(JAVA_SOURCES) -d .
+	@javac -target 1.8 -classpath $(WEBOTS_HOME_PATH)/lib/java/Controller.jar $(JAVA_SOURCES) -d .
 
 $(JAVA_SOURCES): $(WRAPPER)
 

--- a/projects/default/libraries/vehicle/python/Makefile
+++ b/projects/default/libraries/vehicle/python/Makefile
@@ -15,6 +15,7 @@
 space :=
 space +=
 WEBOTS_HOME_PATH=$(subst $(space),\ ,$(strip $(subst \,/,$(WEBOTS_HOME))))
+WEBOTS_HOME_PATH_NO_ESCAPE=$(subst \,/,$(WEBOTS_HOME))
 include $(WEBOTS_HOME_PATH)/resources/Makefile.os.include
 
 PYTHON_COMMAND ?= python
@@ -25,7 +26,7 @@ WEBOTS_LIBRARY = 1
 MODULE_NAME = vehicle
 
 INTERFACE          = $(MODULE_NAME).i
-SOURCE_HEADERS     = $(wildcard $(WEBOTS_HOME_PATH)/include/controller/cpp/webots/*.hpp)
+SOURCE_HEADERS     = $(substr $(WEBOTS_HOME_PATH_NO_ESCAPE),$(WEBOTS_HOME_PATH),$(wildcard $(WEBOTS_HOME_PATH)/include/controller/cpp/webots/*.hpp))
 SWIG_OPTS          = -c++ -python -outdir $(PYTHON_LIB_FOLDER) -I$(WEBOTS_HOME_PATH)/resources/languages/python/
 WRAPPER            = $(MODULE_NAME)$(PYTHON_SHORT_VERSION).cpp
 WRAPPER_OBJECT     = $(WRAPPER:.cpp=.o)

--- a/projects/default/libraries/vehicle/python/Makefile
+++ b/projects/default/libraries/vehicle/python/Makefile
@@ -26,12 +26,12 @@ MODULE_NAME = vehicle
 
 INTERFACE          = $(MODULE_NAME).i
 SOURCE_HEADERS     = $(wildcard $(WEBOTS_HOME_PATH)/include/controller/cpp/webots/*.hpp)
-SWIG_OPTS          = -c++ -python -outdir $(PYTHON_LIB_FOLDER) -I"$(WEBOTS_HOME_PATH)/resources/languages/python/"
+SWIG_OPTS          = -c++ -python -outdir $(PYTHON_LIB_FOLDER) -I$(WEBOTS_HOME_PATH)/resources/languages/python/
 WRAPPER            = $(MODULE_NAME)$(PYTHON_SHORT_VERSION).cpp
 WRAPPER_OBJECT     = $(WRAPPER:.cpp=.o)
 PYOUT              = $(PYTHON_LIB_FOLDER)/$(MODULE_NAME).py
 PYTHON_VERSION    ?= 2.7
-INCLUDES           = -I"$(WEBOTS_HOME_PATH)/include/controller/c" -I"$(WEBOTS_HOME_PATH)/include/controller/cpp"
+INCLUDES           = -I$(WEBOTS_HOME_PATH)/include/controller/c -I$(WEBOTS_HOME_PATH)/include/controller/cpp
 LIB                = -lCppCar -lCppDriver
 
 ifeq ($(OSTYPE),linux)

--- a/projects/robots/mobsya/thymio/libraries/Makefile
+++ b/projects/robots/mobsya/thymio/libraries/Makefile
@@ -12,7 +12,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-include $(WEBOTS_HOME)/resources/Makefile.os.include
+space :=
+space +=
+WEBOTS_HOME_PATH=$(subst $(space),\ ,$(strip $(subst \,/,$(WEBOTS_HOME))))
+include $(WEBOTS_HOME_PATH)/resources/Makefile.os.include
 
 ifeq ($(OSTYPE),windows)
 CMAKE_FLAGS = -G "MSYS Makefiles"

--- a/projects/vehicles/controllers/ros_automobile/Makefile
+++ b/projects/vehicles/controllers/ros_automobile/Makefile
@@ -50,12 +50,12 @@ include/webots_ros/%.h : include/srv/%.srv $(WEBOTS_HOME_PATH)/projects/default/
 	@echo "# generating ROS header" $(notdir $<)
 	$(SILENT)python $(WEBOTS_HOME_PATH)/projects/default/controllers/ros/headersFromSRV.py $< $(WEBOTS_HOME_PATH)/projects/default/controllers/ros/include
 
-INCLUDE = -I"$(WEBOTS_HOME_PATH)/projects/default/controllers/ros" -isystem "$(WEBOTS_HOME_PATH)/projects/default/controllers/ros/include" -I"include"
+INCLUDE = -I$(WEBOTS_HOME_PATH)/projects/default/controllers/ros -isystem $(WEBOTS_HOME_PATH)/projects/default/controllers/ros/include -Iinclude
 LIBRARIES += -lCppDriver -lCppCar -ldriver -lcar
 
 # include ros libraries
 
-LIBRARIES += -L"$(WEBOTS_HOME_PATH)/projects/default/libraries/ros" -lroscpp -lrosconsole -lroscpp_serialization  -lrostime
+LIBRARIES += -L$(WEBOTS_HOME_PATH)/projects/default/libraries/ros -lroscpp -lrosconsole -lroscpp_serialization -lrostime
 ifeq ($(OSTYPE),windows)
  LIBRARIES += -lws2_32
  ifeq ($(MAKECMDGOALS),debug)
@@ -68,7 +68,7 @@ ifeq ($(OSTYPE),windows)
 endif
 
 ifeq ($(OSTYPE),linux)
- LIBRARIES += -Wl,-rpath,"$(WEBOTS_HOME_PATH)/projects/default/libraries/ros"
+ LIBRARIES += -Wl,-rpath,$(WEBOTS_HOME_PATH)/projects/default/libraries/ros
 endif
 
 ifeq ($(OSTYPE),darwin)

--- a/resources/languages/cpp/Makefile
+++ b/resources/languages/cpp/Makefile
@@ -94,7 +94,7 @@ endif
 all debug profile release: $(TARGET)
 
 $(TARGET):$(CPPOBJS) $(LIBCONTROLLER)
-	$(CXX) $(LDFLAGS) $(CPPOBJS) $(SHAREDLIBS) -o $@
+	$(CXX) $(LDFLAGS) $(CPPOBJS) $(SHAREDLIBS) -o "$@"
 
 $(LIBCONTROLLER):
 	@echo "Warning: $(LIBCONTROLLER) doesn't exist, rebuild it first."

--- a/resources/languages/cpp/Makefile
+++ b/resources/languages/cpp/Makefile
@@ -64,15 +64,15 @@ LDFLAGS       = -shared -Wl,--out-implib,$(WEBOTS_HOME_PATH)/msys64/mingw64/lib/
 LIBCONTROLLER = $(WEBOTS_HOME_PATH)/msys64/mingw64/bin/Controller.dll
 SHAREDLIBS    = $(LIBCONTROLLER)
 CPPFLAGS      = -c -Wall -mwindows
-CPPINCLUDES   = -I"$(WEBOTS_HOME_PATH)/include/controller/c" -I"$(WEBOTS_HOME_PATH)/include/controller/cpp"
+CPPINCLUDES   = -I$(WEBOTS_HOME_PATH)/include/controller/c -I$(WEBOTS_HOME_PATH)/include/controller/cpp
 TARGET        = $(WEBOTS_HOME_PATH)/msys64/mingw64/bin/CppController.dll
 endif
 
 ifeq ($(OSTYPE),darwin)
 LDFLAGS       = -dynamiclib -compatibility_version 1.0 -current_version 1.0.0 -mmacosx-version-min=$(MACOSX_MIN_SDK_VERSION)
-SHAREDLIBS    = -L"$(WEBOTS_HOME_PATH)/lib" -lController
+SHAREDLIBS    = -L$(WEBOTS_HOME_PATH)/lib -lController
 CPPFLAGS      = -c -fPIC -Wall -mmacosx-version-min=$(MACOSX_MIN_SDK_VERSION)
-CPPINCLUDES   = -I"$(WEBOTS_HOME_PATH)/include/controller/c" -I"$(WEBOTS_HOME_PATH)/include/controller/cpp"
+CPPINCLUDES   = -I$(WEBOTS_HOME_PATH)/include/controller/c -I$(WEBOTS_HOME_PATH)/include/controller/cpp
 LIBCONTROLLER = $(WEBOTS_HOME_PATH)/lib/libController.dylib
 LDFLAGS      += -install_name @rpath/lib/libCppController.dylib
 TARGET        = $(WEBOTS_HOME_PATH)/lib/libCppController.dylib
@@ -80,9 +80,9 @@ endif
 
 ifeq ($(OSTYPE),linux)
 LDFLAGS       = -shared
-SHAREDLIBS    = -L"$(WEBOTS_HOME_PATH)/lib" -lController
+SHAREDLIBS    = -L$(WEBOTS_HOME_PATH)/lib -lController
 CPPFLAGS      = -c -fpic -Wall
-CPPINCLUDES   = -I"$(WEBOTS_HOME_PATH)/include/controller/c" -I"$(WEBOTS_HOME_PATH)/include/controller/cpp"
+CPPINCLUDES   = -I$(WEBOTS_HOME_PATH)/include/controller/c -I$(WEBOTS_HOME_PATH)/include/controller/cpp
 LIBCONTROLLER = $(WEBOTS_HOME_PATH)/lib/libController.so
 TARGET        = $(WEBOTS_HOME_PATH)/lib/libCppController.so
 endif

--- a/resources/languages/java/Makefile
+++ b/resources/languages/java/Makefile
@@ -76,7 +76,7 @@ LIBNAME         = JavaController
 .PHONY: JAVA_CLASSES
 
 #compilation options
-WEBOTS_INCLUDES = -I"$(WEBOTS_HOME_PATH)/include/controller/cpp" -I"$(WEBOTS_HOME_PATH)/include/controller/c"
+WEBOTS_INCLUDES = -I$(WEBOTS_HOME_PATH)/include/controller/cpp -I$(WEBOTS_HOME_PATH)/include/controller/c
 
 SWIG_EXISTS = $(shell which $(SWIG) 2> /dev/null)
 
@@ -86,7 +86,7 @@ endif
 
 # variables depending on the OS
 ifeq ($(OSTYPE),windows)
-CFLAGS1         = -I"$(WEBOTS_HOME_PATH)/windows/include"
+CFLAGS1         = -I$(WEBOTS_HOME_PATH)/windows/include
 PLATFORM_INCLUDE= -I"$(JAVA_HOME)/include/win32"
 PLATFORM_LIBS   =
 TARGET          = $(WEBOTS_HOME_PATH)/lib/java/$(LIBNAME).dll
@@ -142,17 +142,17 @@ $(WRAPPER_OBJECT): $(WRAPPER)
 
 $(WEBOTS_HOME_PATH)/lib/java/lib$(LIBNAME).so: $(WRAPPER_OBJECT)
 	@mkdir -p $(WEBOTS_HOME_PATH)/lib/java
-	$(CXX) -shared $^ -L"$(WEBOTS_HOME_PATH)/lib" -lController -lCppController -o $@
+	$(CXX) -shared $^ -L$(WEBOTS_HOME_PATH)/lib -lController -lCppController -o $@
 	@chmod a-x $@
 
 $(WEBOTS_HOME_PATH)/lib/java/lib$(LIBNAME).jnilib: $(WRAPPER_OBJECT)
 	@mkdir -p $(WEBOTS_HOME_PATH)/lib/java
-	$(CXX) $(CFLAGS1) $(JNILIB_FLAGS) -L"$(WEBOTS_HOME_PATH)/lib" -lController -lCppController $^ -o $@
+	$(CXX) $(CFLAGS1) $(JNILIB_FLAGS) -L$(WEBOTS_HOME_PATH)/lib -lController -lCppController $^ -o $@
 
 $(WEBOTS_HOME_PATH)/lib/java/$(LIBNAME).dll: $(WRAPPER_OBJECT)
 	@mkdir -p $(WEBOTS_HOME_PATH)/lib/java
 	$(CXX) -shared -mwindows -Wl,--add-stdcall-alias -Wl,--enable-auto-import -O \
-	$^ -L"$(WEBOTS_HOME_PATH)/msys64/mingw64/bin" -lController -lCppController -lm -o $@
+	$^ -L$(WEBOTS_HOME_PATH)/msys64/mingw64/bin -lController -lCppController -lm -o $@
 
 $(JAR_FILE): JAVA_CLASSES
 	@mkdir -p $(WEBOTS_HOME_PATH)/lib/java

--- a/resources/languages/java/Makefile
+++ b/resources/languages/java/Makefile
@@ -147,22 +147,22 @@ $(WEBOTS_HOME_PATH)/lib/java/lib$(LIBNAME).so: $(WRAPPER_OBJECT)
 
 $(WEBOTS_HOME_PATH)/lib/java/lib$(LIBNAME).jnilib: $(WRAPPER_OBJECT)
 	@mkdir -p $(WEBOTS_HOME_PATH)/lib/java
-	$(CXX) $(CFLAGS1) $(JNILIB_FLAGS) -L$(WEBOTS_HOME_PATH)/lib -lController -lCppController $^ -o $@
+	$(CXX) $(CFLAGS1) $(JNILIB_FLAGS) -L$(WEBOTS_HOME_PATH)/lib -lController -lCppController $^ -o "$@"
 
 $(WEBOTS_HOME_PATH)/lib/java/$(LIBNAME).dll: $(WRAPPER_OBJECT)
 	@mkdir -p $(WEBOTS_HOME_PATH)/lib/java
 	$(CXX) -shared -mwindows -Wl,--add-stdcall-alias -Wl,--enable-auto-import -O \
-	$^ -L$(WEBOTS_HOME_PATH)/msys64/mingw64/bin -lController -lCppController -lm -o $@
+	$^ -L$(WEBOTS_HOME_PATH)/msys64/mingw64/bin -lController -lCppController -lm -o "$@"
 
 $(JAR_FILE): JAVA_CLASSES
 	@mkdir -p $(WEBOTS_HOME_PATH)/lib/java
-	jar cfv $@ $(CONTROLLERPATH)/*.class > /dev/null
+	jar cfv $(JAR_FILE) $(CONTROLLERPATH)/*.class > /dev/null
 
 $(JAVA_SOURCES): $(WRAPPER)
 
 $(WRAPPER): $(INTERFACE) $(LIBCONTROLLER) $(LIBCPPCONTROLLER)
 	test -d $(SWIG_OUTPUT_PATH) || mkdir -p $(SWIG_OUTPUT_PATH)
-	$(SWIG) -c++ -java -package $(subst /,.,$(CONTROLLERPATH)) -outdir $(SWIG_OUTPUT_PATH) $(WEBOTS_INCLUDES) -o $(WRAPPER) $<
+	$(SWIG) -c++ -java -package $(subst /,.,$(CONTROLLERPATH)) -outdir $(SWIG_OUTPUT_PATH) $(WEBOTS_INCLUDES) -o "$@" $<
 
 $(LIBCONTROLLER):
 	@echo "$(LIBCONTROLLER) doesn't exist, build it first."

--- a/resources/languages/python/Makefile
+++ b/resources/languages/python/Makefile
@@ -26,8 +26,8 @@ PYTHON_VERSION := $(shell $(PYTHON_COMMAND) --version 2>&1 | sed 's/.* \([0-9]\)
 PYTHON_SHORT_VERSION = $(subst .,,$(PYTHON_VERSION))
 INTERFACE       = controller.i
 SWIG            = swig
-SWIG_OPTS       = -c++ -python -outdir "$(WEBOTS_HOME_PATH)/lib/python$(PYTHON_SHORT_VERSION)/"
-WEBOTS_INCLUDES = -I"$(WEBOTS_HOME_PATH)/include/controller/cpp" -I"$(WEBOTS_HOME_PATH)/include/controller/c"
+SWIG_OPTS       = -c++ -python -outdir $(WEBOTS_HOME_PATH)/lib/python$(PYTHON_SHORT_VERSION)/
+WEBOTS_INCLUDES = -I$(WEBOTS_HOME_PATH)/include/controller/cpp -I$(WEBOTS_HOME_PATH)/include/controller/c
 WRAPPER         = $(INTERFACE:.i=$(PYTHON_SHORT_VERSION).cpp)
 WRAPPER_OBJECT  = $(WRAPPER:.cpp=.o)
 PYOUT           = $(addprefix $(WEBOTS_HOME_PATH)/lib/python$(PYTHON_SHORT_VERSION)/,$(INTERFACE:.i=.py))
@@ -39,7 +39,7 @@ SPACE          +=
 PYTHON_HOME    := $(subst \ ,$(SPACE),$(dir $(subst $(SPACE),\ ,$(shell which python 2> /dev/null))))
 C_FLAGS         = -c -O -Wall -DMS_WIN64 -D_hypot=hypot -Wno-stringop-truncation
 LD_FLAGS        = -shared -Wl,--enable-auto-import
-LIBS            = -L"$(PYTHON_HOME)libs" -lpython$(PYTHON_SHORT_VERSION) -L"$(WEBOTS_HOME_PATH)/msys64/mingw64/bin" -lController -lCppController
+LIBS            = -L"$(PYTHON_HOME)libs" -lpython$(PYTHON_SHORT_VERSION) -L$(WEBOTS_HOME_PATH)/msys64/mingw64/bin -lController -lCppController
 LIBOUT          = $(addprefix $(WEBOTS_HOME_PATH)/lib/python$(PYTHON_SHORT_VERSION)/_,$(INTERFACE:.i=.pyd))
 ifneq (,$(findstring 3.,$(PYTHON_VERSION)))
 DEF             = _controller3.def
@@ -65,7 +65,7 @@ ifeq ($(findstring llvm-g++,$(shell ls -lF $(shell which c++ 2> /dev/null))),)
 C_FLAGS        += -Wno-self-assign
 endif
 LD_FLAGS        = -dynamiclib -install_name @rpath/lib/python$(PYTHON_SHORT_VERSION)/_$(INTERFACE:.i=.dylib) -Wl,-rpath,@loader_path/../.. -compatibility_version 1.0 -current_version 1.0.0 -mmacosx-version-min=$(MACOSX_MIN_SDK_VERSION)
-LIBS            = -L"$(PYTHON_PATH)/lib" -L"$(WEBOTS_HOME_PATH)/lib" -lController -lCppController -lpython$(PYTHON_VERSION)
+LIBS            = -L"$(PYTHON_PATH)/lib" -L$(WEBOTS_HOME_PATH)/lib -lController -lCppController -lpython$(PYTHON_VERSION)
 LIBOUT          = $(addprefix $(WEBOTS_HOME_PATH)/lib/python$(PYTHON_SHORT_VERSION)/_,$(INTERFACE:.i=.so))
 PYTHON_INCLUDES = -I"$(PYTHON_PATH)/include/python$(PYTHON_VERSION)$(PYTHON_PYMALLOC)"
 LIBCONTROLLER   = $(WEBOTS_HOME_PATH)/lib/libController.dylib
@@ -75,7 +75,7 @@ endif
 ifeq ($(OSTYPE),linux)
 C_FLAGS         = -c -Wall -fPIC -Wno-unused-but-set-variable
 LD_FLAGS        = -shared
-LIBS            = -L"$(WEBOTS_HOME_PATH)/lib" -lController -lCppController
+LIBS            = -L$(WEBOTS_HOME_PATH)/lib -lController -lCppController
 LIBOUT          = $(addprefix $(WEBOTS_HOME_PATH)/lib/python$(PYTHON_SHORT_VERSION)/_,$(INTERFACE:.i=.so))
 PYTHON_INCLUDES = -I"/usr/include/python$(PYTHON_VERSION)"
 LIBCONTROLLER   = $(WEBOTS_HOME_PATH)/lib/libController.so

--- a/resources/osm_importer/osm_gui/Makefile
+++ b/resources/osm_importer/osm_gui/Makefile
@@ -12,13 +12,16 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Webots Makefile system 
+# Webots Makefile system
 #
 # You may add some variable definitions hereafter to customize the build process
 # See documentation in $(WEBOTS_HOME_PATH)/resources/Makefile.include
 
 ### Do not modify: this includes Webots global Makefile.include
-include $(WEBOTS_HOME)/resources/Makefile.os.include
+space :=
+space +=
+WEBOTS_HOME_PATH=$(subst $(space),\ ,$(strip $(subst \,/,$(WEBOTS_HOME))))
+include $(WEBOTS_HOME_PATH)/resources/Makefile.os.include
 
 QT = core gui widgets network
 
@@ -31,7 +34,4 @@ endif
 
 CURDIR = controllers/osm_gui # hawful hack to generate an executable outside of the 'controllers' directory
 
-space :=
-space +=
-WEBOTS_HOME_PATH=$(subst $(space),\ ,$(strip $(subst \,/,$(WEBOTS_HOME))))
 include $(WEBOTS_HOME_PATH)/resources/Makefile.include

--- a/resources/projects/plugins/physics/Makefile
+++ b/resources/projects/plugins/physics/Makefile
@@ -12,7 +12,15 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-include ../../../Makefile.os.include
+ifeq ($(WEBOTS_HOME),)
+WEBOTS_HOME_PATH = ../../..
+else
+space :=
+space +=
+WEBOTS_HOME_PATH=$(subst $(space),\ ,$(strip $(subst \,/,$(WEBOTS_HOME))))
+endif
+
+include $(WEBOTS_HOME_PATH)/resources/Makefile.os.include
 
 CFLAGS = -c -O3
 
@@ -26,8 +34,8 @@ ifeq ($(OSTYPE), linux)
 PLATFORM_FLAGS = -fPIC
 endif
 
-physics.o: physics.c $(WEBOTS_HOME)/include/plugins/physics.h
-	@gcc $(CFLAGS) $(PLATFORM_FLAGS) physics.c -I$(WEBOTS_HOME)/include -I$(WEBOTS_HOME)/include/ode
+physics.o: physics.c $(WEBOTS_HOME_PATH)/include/plugins/physics.h
+	@gcc $(CFLAGS) $(PLATFORM_FLAGS) physics.c -I$(WEBOTS_HOME_PATH)/include -I$(WEBOTS_HOME_PATH)/include/ode
 release debug profile: physics.o
 clean:
 	@rm -f physics.o

--- a/src/packaging/Makefile
+++ b/src/packaging/Makefile
@@ -48,7 +48,7 @@ endif
 endif
 
 ifeq ($(OSTYPE),windows)
-LIBRARIES = -L"$(WEBOTS_HOME_PATH)/lib" -L/mingw64/bin -lcrypto
+LIBRARIES = -L$(WEBOTS_HOME_PATH)/lib -L/mingw64/bin -lcrypto
 WEBOTS_DISTRO_EXE = webots_distro.exe
 INNO_SETUP_HOME ?= /C/Program Files (x86)/Inno Setup 5
 WEBOTS = webots

